modules/auth/oauth2/oauth2.go

```diff
@@ -0,0 +1,57 @@
+package oauth2
+
+import (
+	"code.gitea.io/gitea/modules/setting"
+	"github.com/gorilla/sessions"
+	"github.com/markbates/goth"
+	"github.com/markbates/goth/gothic"
+	"github.com/markbates/goth/providers/github"
+	"net/http"
+	"os"
+	"github.com/satori/go.uuid"
+)
+
+var (
+	sessionUsersStoreKey = "gitea-oauth-sessions"
+	providerHeaderKey = "gitea-oauth-provider"
+)
+
+func init() {
+	gothic.Store = sessions.NewFilesystemStore(os.TempDir(), []byte(sessionUsersStoreKey))
```

Comment:
changed it to use a new data/sessions/oauth2 folder based on the setting.WorkDir method